tfe/resource_tfe_notification_configuration_test.go

```diff
@@ -0,0 +1,436 @@
+package tfe
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccTFENotificationConfiguration_basic(t *testing.T) {
+	notificationConfiguration := &tfe.NotificationConfiguration{}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFENotificationConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFENotificationConfiguration_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFENotificationConfigurationExists(
+						"tfe_notification_configuration.foobar", notificationConfiguration),
+					testAccCheckTFENotificationConfigurationAttributes(notificationConfiguration),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "destination_type", "generic"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "name", "notification_basic"),
+					// Just test the number of items in triggers
+					// Values in triggers attribute are tested by testCheckTFENotificationConfigurationAttributes
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "triggers.#", "0"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "url", "http://example.com"),
+				),
+			},
+			{
+				ResourceName:            "tfe_notification_configuration.foobar",
+				ImportState:             true,
```

Comment:
The import test here should be spun off into another test, unless it needs to be in basic for some reason?

Reply:
Agree. It should probably have just one dedicated import test.

Reply:
I was going to make this a dedicated import test but when reading through [the docs on imports](https://www.terraform.io/docs/extend/resources/import.html), it recommended against that. Did I misunderstand that or are the docs out of date?

Reply:
I kind of wonder if maybe the example needs some clarification? I've never seen the import test combined like that. The `existing TestStep ` part may also be pertaining to what you would need to do to set up an import case.

Reply:
I went ahead and pulled it out into its own test. That's consistent with the rest of the provider. 

I'm making a note to investigate this more later though, because under [the "Additional Information" section in the docs](https://www.terraform.io/docs/extend/resources/import.html#additional-information), it seems to explicitly says you shouldn't make a separate import test.

Reply:
@lafentres ah, yeah that lays it out much more clearly. This could have changed since I was writing providers on a regular basis, so I would say definitely go by the document in the future! The enablement team would probably be the best people to talk here about this, and is the team that probably put this in.